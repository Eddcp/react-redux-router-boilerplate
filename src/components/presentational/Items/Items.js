import React from 'react';
import PropTypes from 'prop-types';

/**
 * Render a list of items
 *
 * @param {Object} props - List of items
 */
function Items(props) {
  const { items = [] } = props;

  if (!items.length) {
    return <span>No items in list</span>;
  }

  if (items.length === 1) {
    return <span>{items[0]}</span>;
  }

  return (
    <ul>
      {items.map(item => <li key={item}>{item}</li>)}
    </ul>
  );
}

Items.propTypes = {
  items: PropTypes.array,
};

Items.defaultProps = {
  items: [],
};

export default Items;