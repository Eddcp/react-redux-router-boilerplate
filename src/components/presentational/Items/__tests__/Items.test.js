import React from 'react';
import renderer from 'react-test-renderer';

import Items from '../Items';

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<Items />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when there is one item', () => {
    const items = ['one'];
    const tree = renderer.create(<Items items={items} />).toJSON();
    expect(tree).toMatchSnapshot();
});
  
it('renders correctly when there are multiple items', () => {
const items = ['one', 'two', 'three'];
const tree = renderer.create(<Items items={items} />).toJSON();
expect(tree).toMatchSnapshot();
});