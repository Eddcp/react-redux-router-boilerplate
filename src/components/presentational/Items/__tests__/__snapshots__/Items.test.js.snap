// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are multiple items 1`] = `
<ul>
  <li>
    one
  </li>
  <li>
    two
  </li>
  <li>
    three
  </li>
</ul>
`;

exports[`renders correctly when there are no items 1`] = `
<span>
  No items in list
</span>
`;

exports[`renders correctly when there is one item 1`] = `
<span>
  one
</span>
`;
